--- a/board/mediatek/mt7988/mt7988_rfb.c
+++ b/board/mediatek/mt7988/mt7988_rfb.c
@@ -53,6 +53,12 @@
 	const char *media;
 	u32 media_handle;
 
+	/* check if NVMe boot was requested via env */
+	if (env_get("nvme_boot")) {
+		media = "rootdisk-nvme";
+		goto set_rootdisk;
+	}
+
 	switch ((readl(0x1001f6f0) & 0xc00) >> 10) {
 	case MT7988_BOOT_NOR:
 		media = "rootdisk-nor";
@@ -71,7 +77,7 @@
 		break
 		;;
 	}
-
+set_rootdisk:
 	chosen = fdt_path_offset(blob, "/chosen");
 	if (chosen <= 0)
 		return 0;
